src/core_plugins/kibana/public/management/sections/indices/_create.js

```diff
@@ -247,6 +254,19 @@ uiModules.get('apps/management')
     }, notify.fatal);
   }
 
+  function updateFieldListAndSeed(results, seedField) {
+
+    updateFieldList(results);
+
+    if (!results.dateFields.length) {
+      return;
+    }
+
+    const matchingField = results.dateFields.find(field => field.name === seedField.name);
+    index.timeField = matchingField ? matchingField : results.dateFields[0];
+
+  }
```

Comment:
Also, I think we should tweak the names here to reinforce the intent of seeding the date field. As it is, it isn't clear that this function is specific to date fields unless you read it's implementation and get to line 265.

I think we can make the logic a little clearer by renaming `matchingField` to something which reflects its role as a flag, because we don't really want to set it as the `timeField` value, we want to set the value passed into the function.

```javascript
  function updateFieldListAndSetTimeField(results, timeField) {
    updateFieldList(results);

    if (!results.dateFields.length) {
      return;
    }

    const defaultTimeField = results.dateFields[0];
    const isTimeFieldAvailable = results.dateFields.find(field => field.name === timeField.name);
    index.timeField = isTimeFieldAvailable ? timeField : defaultTimeField;
  }
```

Lastly, how about combining this with `updateFieldList` and using `timeField` as an optional argument? Then you won't need a condition inside of `refreshFieldList`:

```javascript
  function updateFieldList(results, timeField = undefined) {
    index.fetchFieldsError = results.fetchFieldsError;
    index.dateFields = results.dateFields;

    if (!timeField) {
      return;
    }

    if (!results.dateFields.length) {
      return;
    }

    const defaultTimeField = results.dateFields[0];
    const isTimeFieldAvailable = results.dateFields.find(field => field.name === timeField.name);
    index.timeField = isTimeFieldAvailable ? timeField : defaultTimeField;
  }
```